tests/test_assert.h

```diff
@@ -0,0 +1,39 @@
+#ifndef TEST_ASSERT_H
```

Comment:
ahh yes, nice to share this code.

Reply:
Does this comment mean, you now prefer separation of test code in .h and .cpp?

Reply:
No, this meant it is nice this code is shared between the main test and the gRPC test.
I still prefer things to be in headers, but I can be convinced otherwise.